### PR TITLE
feat: Migrate security settings to project-scoped .uloop/settings.security.json

### PR DESCRIPTION
**Pull request by hatayama**
## Summary

- Migrated 4 security settings (`enableTestsExecution`, `allowMenuItemExecution`, `allowThirdPartyTools`, `dynamicCodeSecurityLevel`) from `UserSettings/UnityMcpSettings.json` to `.uloop/settings.security.json`
- The new file is stored in the project root under `.uloop/`, enabling git-tracked team-wide security policy sharing
- Implemented one-time migration from legacy format with a probe class pattern that safely extracts security fields from the old JSON
- Fixed a race condition where `McpEditorSettings.LoadSettings()` could strip security fields from the legacy file before `ULoopSettings` migration runs

### Why .uloop/settings.security.json?

The previous location (`UserSettings/UnityMcpSettings.json`) is user-specific and not git-tracked, making it impossible to share security policies across team members. The new location allows teams to commit security settings to version control.

### Migration Race Condition Fix

During Unity domain reload, `McpEditorSettings.LoadSettings()` may be called before `ULoopSettings` has a chance to migrate. Since `McpEditorSettingsData` no longer defines security fields, any `SaveSettings()` call re-serializes the JSON without them — permanently losing security values. The fix calls `ULoopSettings.GetSettings()` inside `McpEditorSettings.LoadSettings()` immediately after cache initialization, ensuring migration happens before any `SaveSettings()` can strip the fields.

### Key files changed

- `McpConstants.cs`: Added `ULOOP_SETTINGS_FILE_NAME` constant
- `ULoopSettingsData.cs`: New `record` type for security settings
- `ULoopSettings.cs`: New settings manager with load/save/migration logic
- `McpEditorSettings.cs`: Removed security fields from data class, added migration trigger in `LoadSettings()`
- `ULoopSettingsTests.cs`: 9 tests covering migration, round-trip, and race condition scenarios

Related to #635

## Test Plan

- [x] 9 unit tests pass (migration, round-trip, edge cases, race conditions)
- [x] TDD Red-Green verified: reverting the fix causes Test 8 and 9 to fail
- [x] Unity compilation succeeds with no errors or warnings

---
## Summary by cubic
Migrated four security settings to a project-scoped .uloop/settings.security.json for git-tracked, team-wide policies. Adds a safe one-time migration from UserSettings and fixes a domain reload race that could drop these values.

- **New Features**
  - Project-scoped .uloop/settings.security.json managed by ULoopSettings; tools and UI now read/write via this API.
  - One-time migration from UserSettings/UnityMcpSettings.json with idempotency, .bak recovery, and legacy file re-save that removes security fields while keeping others.
  - Extracted AtomicFileWriter for atomic settings writes.

- **Bug Fixes**
  - Prevented loss of security fields by triggering ULoopSettings.GetSettings() during McpEditorSettings.LoadSettings() before any SaveSettings() can strip legacy data.
  - Added 9 tests covering migration, idempotency, round-trip, backup restore, and domain reload race scenarios.
  - Dependencies: bumped minimatch override to 10.2.4 in CLI and TypeScriptServer.

<sup>Written for commit 10805fc3618db6f2499c65f64265104f0111107a. Summary will update on new commits.</sup>

## Overview
This PR migrates four security settings (enableTestsExecution, allowMenuItemExecution, allowThirdPartyTools, dynamicCodeSecurityLevel) from the per-user file UserSettings/UnityMcpSettings.json to a project-scoped file at .uloop/settings.security.json, enabling teams to track security policy in version control and configure settings per-project rather than per-user.

### New Infrastructure
- ULoopSettings.cs: New static configuration manager handling .uloop/settings.security.json with atomic writes, backup rotation, and automatic recovery from .bak files.
- ULoopSettingsData.cs: Serializable record defining the project-scoped security settings structure.
- AtomicFileWriter.cs: Extracted utility class providing atomic file writing with .tmp/.bak rotation to minimize exposure to partial writes.

### Migration & Configuration
- Automatic legacy migration: One-time, idempotent migration from legacy McpEditorSettings.json to .uloop/settings.security.json using a LegacySecuritySettingsProbe pattern; creates .bak recovery files as needed.
- McpConstants.cs: Added ULOOP_DIR and ULOOP_SETTINGS_FILE_NAME constants; updated error messages to reference the new settings location.
- McpEditorSettings.cs: Removed security-related fields and public getters/setters; added a migration trigger (ULoopSettings.GetSettings()) in LoadSettings() to ensure migration runs before any SaveSettings() that might strip fields; added InvalidateCache() entry point.

### Roslyn Integration
- ULoopSettings manages the SCRIPTING_DEFINE_ULOOPMCP_HAS_ROSLYN scripting define based on dynamicCodeSecurityLevel, with per-build-target checks and re-evaluation on domain reload.

### API Migration
Code paths changed to read/write security settings from ULoopSettings instead of McpEditorSettings:
- ExecuteDynamicCodeTool.cs
- McpSecurityChecker.cs
- DangerousApiDetector.cs
- McpEditorModel.cs
- McpEditorWindow.cs
- SecuritySettingsSection.cs
- Relevant tests updated to use ULoopSettings

## Architecture

```mermaid
classDiagram
    class McpConstants {
        +ULOOP_DIR: string
        +ULOOP_SETTINGS_FILE_NAME: string
    }
    
    class ULoopSettingsData {
        +enableTestsExecution: bool
        +allowMenuItemExecution: bool
        +allowThirdPartyTools: bool
        +dynamicCodeSecurityLevel: int
    }
    
    class ULoopSettings {
        -_cachedSettings: ULoopSettingsData
        +GetSettings() ULoopSettingsData
        +SaveSettings(settings) void
        +UpdateSettings(transform) void
        +GetEnableTestsExecution() bool
        +SetEnableTestsExecution(bool) void
        +GetAllowMenuItemExecution() bool
        +SetAllowMenuItemExecution(bool) void
        +GetAllowThirdPartyTools() bool
        +SetAllowThirdPartyTools(bool) void
        +GetDynamicCodeSecurityLevel() DynamicCodeSecurityLevel
        +SetDynamicCodeSecurityLevel(DynamicCodeSecurityLevel) void
        -MigrateFromLegacySettings() void
        -UpdateRoslynSymbol() void
    }
    
    class AtomicFileWriter {
        +Write(path, content) void
        +CleanupBackup(backupPath) void
    }
    
    class McpEditorSettings {
        -_cachedSettings: McpEditorSettingsData
        +InvalidateCache() void
        +LoadSettings() void
        +SaveSettings(settings) void
    }
    
    ULoopSettings --> ULoopSettingsData
    ULoopSettings --> AtomicFileWriter
    McpEditorSettings --> AtomicFileWriter
    McpConstants --|> ULoopSettings
```

## Race Condition Fix
Addresses a domain-reload race where McpEditorSettings.LoadSettings() could run before migration and a subsequent SaveSettings() would re-serialize the legacy file without security fields. The fix triggers ULoopSettings.GetSettings() early in McpEditorSettings.LoadSettings() so migration runs before any potential save.

## Testing
- Assets/Tests/Editor/ULoopSettingsTests.cs: NUnit suite with 9 tests covering migration idempotency, round-trip read/write, precedence and cleanup behavior, defaults when files are absent, cache invalidation, recovery from .bak backups, and simulated domain-reload race conditions.
- Assets/Tests/Editor/DynamicCodeToolTests/ExecuteDynamicCodeParameterValidationTests.cs: updated to use ULoopSettings for dynamic security level access.
- All included tests pass in the verification run; reverting the race-fix breaks tests that validate migration/race behavior.

## Files Changed Summary
- Added: Packages/src/Editor/Config/ULoopSettings.cs, Packages/src/Editor/Config/ULoopSettingsData.cs, Packages/src/Editor/Config/AtomicFileWriter.cs
- Modified: Packages/src/Editor/Config/McpEditorSettings.cs, Packages/src/Editor/Config/McpConstants.cs, multiple UI and security components to call ULoopSettings, updated tests under Assets/Tests/Editor
- Tests: Assets/Tests/Editor/ULoopSettingsTests.cs added (9 tests); ExecuteDynamicCodeParameterValidationTests updated

### CODE REVIEW

File: Packages/src/Editor/Config/ULoopSettings.cs [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+        private static void LoadSettings()
+        {
+            // Recover from interrupted atomic write
+            string backupPath = SettingsFilePath + ".bak";
+            if (!File.Exists(SettingsFilePath) && File.Exists(backupPath))
+            {
+                File.Move(backupPath, SettingsFilePath);
+            }
+
+            if (File.Exists(SettingsFilePath))
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Editor/Config/ULoopSettings.cs:138]: Fixed. Wrapped `LoadSettings()` in try-catch to match `McpEditorSettings.LoadSettings()` pattern. On any file I/O or JSON parse error, the exception is re-thrown as `InvalidOperationException` with context — consistent with how `McpEditorSettings` handles the same scenario.
